Support overhaul (#34)
OctopusDeploy/People: Support.md

# Support Roles

## Junior Support Engineer

_I am starting my journey at Octopus. I am an enthusiastic learner, I show great potential and have a high growth trajectory toward independent contribution._

> >We avoid distinguishing too much between Junior Support Engineer and Support Engineer. These roles are primarily focused on learning with the aim to grow into the contribution of a Senior Support Engineer or beyond. The primary indicator inside these levels is how far you are along your journey from dependent contribution to independent contribution.

Reviewer: I don't know if this distinction is helpful or not, but I've always thought of the difference between "junior role" and "role", as a junior being somebody who is actively learning how to do the role and needs support, training, feedback, mentoring, while somebody in a non-junior role also needs those to continue to grow, they don't need it as much to be independently productive.

### I aim to write high quality responses quickly and with minimal oversight
- I recognize Octopus is complex and I am starting on my knowledge journey, I reach out for help in a timely and effective way when I need it.
- I am learning to write well crafted, succinct and informative support responses with coaching from my mentor.

> - I have started my learning journey in my area specialty at Octopus, I am seeking to be able to independently answer support questions in this area.

Reviewer: ```suggestion
- I have started my learning journey in my area speciality at Octopus, I am seeking to be able to independently answer support questions in this area.
```

- I am learning to write clear, well researched issues that provide the Engineering team with clear customer impact and assist in root cause analysis.
- I am becoming proficient with my tools and the core building blocks used by our team.
- I am learning to prioritise my workload.
- I am learning to recognise when I need help, and how to do this in an effective and timely manner.
- I am a self-managing team member, I make and keep my commitments using the privilege of flexibility wisely.

### I am recognised by my desire to learn and my motivation to improve

- I take my individual growth seriously and realise it is valuable to both myself and to Octopus.

> - I take the privilege of Sharpening seriously, and work with my Manager to set realistic, achievable and relevant learning goals.

Reviewer: ```suggestion
- I take the privilege of Sharpening seriously, and work with my manager to set realistic, achievable and relevant learning goals.
```

- I am an active learner. I use a variety of learning methods, like reading books, reading code, configuring test environments, watching online courses, attending conferences or user group meetings to help improve my knowledge of Octopus and surrounding technology.
- I am discussing the things I'm learning about or experimenting on with my teammates and manager, maybe even collaborating on the experience.

> - I recognise I am my career owner. I reach out to my peers and my manager to develop my personal growth plan.

Reviewer: ```suggestion
- I recognise I own my career. I reach out to my peers and my manager to develop my personal growth plan.
```

### I am an egoless, team-oriented support engineer
- Other team members genuinely want to work alongside me.
- I happily take responsibility for what I produce.

> - I readily accept input from others, and understand that good ideas can come from any source.

Reviewer: ```suggestion
- I readily accept input from others, and I understand that good ideas can come from any source.
```

- I write well crafted, succinct and informative support responses with minimal coaching.
- I am gaining a good understanding of my area of speciality at Octopus, and can confidently answer support questions in this some of the knowledge areas that make up my speciality.
- I am learning to anticipate my customer's needs. I am learning how to lead our customers to finding their own solutions.
- I am learning how to qualify support tickets into the RUF and KUE categories.
- I am learning to write clear, well researched issues that assist provide the Engineering team with clear customer impact and assist in root cause analysis.
- I am becoming proficient with my tools and the core building blocks used by our team.
- I am learning to prioritise my workload.
- I am learning to recognise when I need help, and how to do this in an effective and timely manner.
- I am a self-managing team member, I make and keep my commitments using the privilege of flexibility wisely.

### I am recognised by my desire to learn and my motivation to improve
- I take my individual growth seriously and realise it is valuable to both myself and to Octopus.

> - I take the privilege of Sharpening seriously, and work with my Manager to set realistic, achievable and relevant learning goals.

Reviewer: ```suggestion
- I take the privilege of Sharpening seriously, and work with my manager to set realistic, achievable and relevant learning goals.
```

- I possess writing skills that let me be clear, concise, articulate, and persuasive.

### I've become a student of my craft
- I am devoted to learning; it's become a natural part of my life.
- I am actively introspective and take time to work on my personal and professional development.
- I happily take responsibility for my own mistakes and I learn from the mistakes of others.

## Lead Support Engineer

_I am a well-rounded support engineer and problem solver. I can be trusted to take any issue through to resoltion, bringing people along with me throughout the process and providing clarity for our team. I am trusted to autonomously lead people through complex problems with a proven record of delivering ideas from inception to their successful completion._

_The Lead Support Engineer role builds upon the foundation of the earlier roles._

### I am trusted
- I have mastered my tools and the core building blocks used by our team.

> - I have a deep technical understanding of my area of specialty at Octopus, and I can confidently use this knowledge to assist customers and fellow support team members in all areas of Octopus.

Reviewer: ```suggestion
- I have a deep technical understanding of my area of speciality at Octopus, and I can confidently use this knowledge to assist customers and fellow support team members in all areas of Octopus.
```

### I am a mentor of mentors
- I know when to directly assist with a ticket, and when to encourage others to learn more.
- I am actively involved in the retrospective process, and I am comfortable challenging and coaching my peers. I encourage others to find creative solutions to problems and accept that my way isn't the only way.
- I am always on the lookout for opportunities to teach and show others alternate, cleaner ways to delight our customers in an ego-less manner. 
- I am working to improve our processes, making it easier for everyone in our team to grow and learn.
- I am sharing my on-the-job learning and experiences with others so they can be more effective in their roles, making sure others understand the solution throughout the process.

### Everything is a learning opportunity
- I am constantly learning new technologies and methodologies that relate to Octopus.
- I work with teams outside of Support to further enhance my knowledge
- I happily take responsibility for my own mistakes and I learn from the mistakes of others.
- I share the lessons I learn so the entire team and Octopus can benefit

## Principal Support Engineer

> _I am an influential Support Leader, a person who identifies problems and opportunities relevant to Octopus' strategy, builds consensus to invest in solutions, and guides Support to deliver significant lasting change across Octopus._

Reviewer: I like this distinction. Really helps explain the influence component of thinking and acting in a much broader way.

_The Principal Support Engineer role builds upon the foundation of the earlier Support roles._

### I uncover important problems and opportunities that are critical to the long-term success of Octopus

> - I have a deep understanding of my area of specialty, as well as Octopus products and strategy

Reviewer: ```suggestion
- I have a deep understanding of my area of speciality, as well as Octopus products and strategy
```

> - I collaborate with other Octopus leaders to ensure I am aware of all major initiatives at Octopus and can understand how they will affect our customers and Support teams

Reviewer: ```suggestion
- I collaborate with other Octopus leaders to ensure I am aware of all major initiatives at Octopus and can understand how they will affect our customers and Support teams.
```

> - I have a track record of identifying significant problems, inhibitors and opportunities, and for creating lasting, best-fit solutions that align our strategy and Support efforts

Reviewer: ```suggestion
- I have a track record of identifying significant problems, inhibitors and opportunities, and for creating lasting, best-fit solutions that align our strategy and Support efforts.
```

- Instead of simply patching or extending existing solutions, which may not be fit for purpose anymore, my contributions open up whole new areas of strategic possibility that progress our journey of making Octopus a world-class Support organization.

### I am the custodian of an area of knowledge
- I am comfortable taking on large-scale greenfields and brownfields initiatives both inside and outside of my area of specialty that assist the Support team and our Customers
- I identify people and teams with the necessary skills, work out how to fill gaps, and guide and influence these people and teams collaboratively
- I use a variety of tools and techniques to effectively communicate short and long-term plans and goals to my teams, stakeholders, Octopus senior leadership, and the broader organization
- I use my experience to collaboratively steer teams toward good long-term decisions
- The results of the initiatives I champion are measurable

### I share my knowledge both within the support team, and with the Octopus and wider community

> - I actively influence support teams across Octopus resulting in improved support excellence and overall productivity
> - I actively talk both internally and externally about the lessons we are learning as we improve and expand the support team to clarifying our thinking, reinforce our learning pathways, and to attract customers and potential future employees
> - I have a proven track record of expertly tailoring my communication to the target audience with a result of increased clarity and impact

Reviewer: ```suggestion
- I actively influence support teams across Octopus resulting in improved support excellence and overall productivity.
- I actively talk both internally and externally about the lessons we are learning as we improve and expand the support team to clarifying our thinking, reinforce our learning pathways, and to attract customers and potential future employees.
- I have a proven track record of expertly tailoring my communication to the target audience with a result of increased clarity and impact.
```
